[KEYCLOAK CI] - AuroraDB IT - Run Aurora new database tests on EC2

### Before reporting an issue

- [x] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

ci

### Describe the bug

Run Aurora new database tests on EC2 failed two times with next err:
```
TASK [mvn_remote_runner : Fail if Maven command failed] ************************
Error: : Task failed: Action failed: Maven command failed with return code 1: 
Origin: /home/runner/work/keycloak/keycloak/.github/scripts/ansible/roles/mvn_remote_runner/tasks/run.yml:58:3

56     - "{{ output.files }}"
57
58 - name: Fail if Maven command failed
     ^ column 3
```
and build failure as well:
```
            "[INFO] ",
Error:      "[ERROR] Failures: ",
Error:      "[ERROR]   DatabaseTestSuite.test expected: <2> but was: <1>",
            "[INFO] ",
Error:      "[ERROR] Tests run: 876, Failures: 1, Errors: 0, Skipped: 2",
```
https://github.com/keycloak/keycloak/actions/runs/21577626346/job/62168259152#step:14:1756
https://github.com/keycloak/keycloak/actions/runs/21617134512/job/62298331213#step:14:1653

### Version

latest

### Regression

- [ ] The issue is a regression

### Expected behavior

-

### Actual behavior

-

### How to Reproduce?

-

### Anything else?

_No response_

--- a/tests/base/src/test/java/org/keycloak/tests/admin/AttackDetectionResourceTest.java
+++ b/tests/base/src/test/java/org/keycloak/tests/admin/AttackDetectionResourceTest.java
@@ -18,8 +18,8 @@
 package org.keycloak.tests.admin;
 
 import java.util.Map;
-
 import java.util.concurrent.TimeUnit;
+
 import org.keycloak.admin.client.resource.AttackDetectionResource;
 import org.keycloak.events.admin.OperationType;
 import org.keycloak.events.admin.ResourceType;

--- a/tests/base/src/test/java/org/keycloak/tests/admin/AttackDetectionResourceTest.java
+++ b/tests/base/src/test/java/org/keycloak/tests/admin/AttackDetectionResourceTest.java
@@ -19,6 +19,7 @@ package org.keycloak.tests.admin;
 
 import java.util.Map;
 
+import java.util.concurrent.TimeUnit;
 import org.keycloak.admin.client.resource.AttackDetectionResource;
 import org.keycloak.events.admin.OperationType;
 import org.keycloak.events.admin.ResourceType;
@@ -38,6 +39,7 @@ import org.keycloak.tests.utils.admin.AdminEventPaths;
 
 import org.junit.jupiter.api.Test;
 
+import static org.awaitility.Awaitility.await;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertNotEquals;
 import static org.junit.jupiter.api.Assertions.assertTrue;
@@ -77,6 +79,15 @@ public class AttackDetectionResourceTest {
         oauthClient.doPasswordGrantRequest(testUser2.getUsername(), "invalid");
         oauthClient.doPasswordGrantRequest(testUser2.getUsername(), "invalid");
         oauthClient.doPasswordGrantRequest("nosuchuser", "invalid");
+
+        // Check testUser2 to ensure all failure processing is completed
+        await().atMost(5, TimeUnit.SECONDS)
+                .pollInterval(100, TimeUnit.MILLISECONDS)
+                .untilAsserted(() -> {
+                    Map<String, Object> status = detection.bruteForceUserStatus(testUser2.getId());
+                    assertEquals(2, status.get("numFailures"),
+                            "Waiting for testUser2 processing to complete");
+                });
 
         assertBruteForce(detection.bruteForceUserStatus(testUser.getId()), 2, 1, true, true);
         assertBruteForce(detection.bruteForceUserStatus(testUser2.getId()), 2, 1, true, true);

